nucypher/network/server.py

```diff
@@ -148,106 +147,27 @@ def condition_chains():
-            context = (
-                json.loads(str(decryption_request.context)) or dict()
-            )  # nucypher_core.Context -> str -> dict
-
-        # obtain condition from request
-        condition_lingo = json.loads(
-            str(decryption_request.acp.conditions)
-        )  # nucypher_core.Conditions -> str -> Lingo
-        if not condition_lingo:
-            # this should never happen for CBD - defeats the purpose
-            return Response(
-                "No conditions present for ciphertext - invalid for CBD functionality",
-                status=HTTPStatus.FORBIDDEN,
-            )
-
-        # evaluate the conditions for this ciphertext
-        error = evaluate_condition_lingo(
-            condition_lingo=condition_lingo,
-            context=context,
-            providers=this_node.condition_providers,
-        )
-        if error:
-            return Response(error.message, status=error.status_code)
-
-        participants = [p.provider for p in ritual.participants]
-
-        # enforces that the node is part of the ritual
-        if this_node.checksum_address not in participants:
-            return Response(
-                f"Node not part of ritual {decryption_request.ritual_id}",
-                status=HTTPStatus.FORBIDDEN,
+            response = Response(
+                response=bytes(encrypted_response),
+                status=HTTPStatus.OK,
+                mimetype="application/octet-stream",
             )
-
-        # derive the decryption share
-        decryption_share = this_node.derive_decryption_share(
-            ritual_id=decryption_request.ritual_id,
-            ciphertext_header=decryption_request.ciphertext_header,
-            aad=decryption_request.acp.aad(),
-            variant=decryption_request.variant,
-        )
-
-        # return the decryption share
-        # TODO: #3098 nucypher-core#49 Use DecryptionShare type
-        decryption_response = ThresholdDecryptionResponse(
-            ritual_id=decryption_request.ritual_id,
-            decryption_share=bytes(decryption_share),
-        )
-        encrypted_response = this_node.encrypt_threshold_decryption_response(
-            decryption_response=decryption_response,
-            requester_public_key=encrypted_decryption_request.requester_public_key,
-        )
-        return Response(
-            bytes(encrypted_response),
-            headers={"Content-Type": "application/octet-stream"},
-        )
+            return response
+        except this_node.RitualNotFoundException:
+            return Response("Ritual not found", status=HTTPStatus.NOT_FOUND)
+        except this_node.UnauthorizedRequest as e:
+            return Response(str(e), status=HTTPStatus.UNAUTHORIZED)
+        except this_node.DecryptionFailure as e:
+            return Response(str(e), status=HTTPStatus.BAD_REQUEST)
+        except Exception as e:
+            return Response(str(e), status=HTTPStatus.INTERNAL_SERVER_ERROR)
```

Comment:
You've convinced me that `DecryptionFailure` should be a 500.

```suggestion
        except this_node.DecryptionFailure as e:
            return Response(str(e), status=HTTPStatus.INTERNAL_SERVER_ERROR)
        except Exception as e:
            return Response(str(e), status=HTTPStatus.INTERNAL_SERVER_ERROR)
```

Reply:
✔️